compiler/control/OMROptions.cpp

```diff
@@ -1988,6 +1988,15 @@ OMR::Options::jitLatePostProcess(TR::OptionSet *optionSet, void * jitConfig)
       self()->setOption(TR_DisableMethodHandleThunks); // Can't yet transition a MH thunk frame into equivalent interpreter frames
       }
 
+   if (self()->getOption(TR_EnableNextGenHCR) && !self()->getOption(TR_DisableOSR))
+      {
+      self()->setOption(TR_EnableOSR);
+      self()->setOption(TR_EnableHCR);
+      self()->setOption(TR_DisableTM);
+      self()->setDisabled(redundantGotoElimination, true);
+      self()->setDisabled(redundantAsyncCheckRemoval, true);
```

Comment:
the disables need all need comments on why they are required. redundantAsyncCheckRemoval is fundamentally not compatible so has to be off. The need to turn off TM and redundantGotoElimination are more dubious and we should verify what the problem is and document that in the options since turning off TM and goto elimination are undesirable if there isn't a good moral reason for having to do so.